SonarQube.TeamBuild.PreProcessor/Interfaces/IAnalyzerProvider.cs

```diff
@@ -19,6 +20,6 @@ public interface IAnalyzerProvider
         /// </summary>
         /// <param name="projectKey">Identifier for the project being analyzed</param>
         /// <returns>The settings required to configure the build for Roslyn analysis</returns>
-        AnalyzerSettings SetupAnalyzers(ISonarQubeServer server, TeamBuildSettings settings, string projectKey);
+        AnalyzerSettings SetupAnalyzers(ISonarQubeServer server, TeamBuildSettings settings, string projectKey, string projectBranch);
```

Comment:
It looks all these methods that need projectKey now also need projectBranch. So methods are now responsible for performing an additional piece of logic (computing the project ID) instead of focusing on a single responsibility 

Would it be possible to hide this complexity by hiding project Key and project Branch and exposing only Project Id ?